TESTING.md

```diff
@@ -0,0 +1,114 @@
+Testing
+=======
+
+Most packages have up to four common sets of tests. A package called
+`Foo` might have:
+
+- `Foo.Tests`: unit tests. These should run quickly, in most cases without
+  an internet connection being required.
+- `Foo.IntegrationTests`: integration tests. These check that our
+  packages work correctly with the real API.
+- `Foo.Snippets`: documentation snippets, written and executed as
+  tests. These often double up as integration tests. (If a snippet can
+  test everything you need to know about a call, there's no point in
+  also having an integration test.) For gRPC APIs, this project
+  contains autogenerated as well as handwritten code; the
+  autogenerated code is only used for documentation.
+- `Foo.SmokeTests`: autogenerated tests that ensure we can at least
+  contact the relevant API.
+
+The test types other than unit tests require:
+
+- An internet connection
+- A Google Cloud Platform project
+- Billing enabled
+- The APIs you're testing enabled
+- Common environment variables set as described below
+- Per-API configuration as shown later, in some cases
+
+**Running these tests can cost you money. They execute against
+paid-for Google services.** They can also produce significant chunks of
+network traffic - particularly some of the integration tests. You're
+not recommended to run them against a mobile internet connection
+unless you're sure you know what you're doing.
+
+**These tests could destroy data in the projects they run in, or
+leave test data afterwards.** Most tests create data in a
+well-partitioned way and clean up after themselves, but there could
+be a bug in a test which wipes all the data in the project. It is
+*very strongly* recommended that you only run tests against projects
+which do not contain any production data or services.
+
+Running tests
+=============
+
+The `build.sh` script will run unit tests, and
+`runintegrationtests.sh` will run integration tests and snippets.
+(TODO: smoke tests too...) Consult the scripts for details on how to
+run.
+
+Additionally, you can just use `dotnet test` in any test project,
+optionally specifying a test filter and/or target framework.
+
+Common environment variables
+============================
+
+To run the integration/smoke tests or snippets, set the following
+environment variables:
+
+- `TEST_PROJECT`: The name of the Google Cloud Project to test against
+- `GOOGLE_APPLICATION_CREDENTIALS`: The absolute path to a JSON file
+  containing service credentials for the test project.
+
+Per API configuration
+=====================
+
+Some APIs require a little more setup. If you're not testing an API
+listed below, the instructions above should be all you need.
+
+Google.Cloud.Bigtable.V2
```

Comment:
It may make sense to also list the Admin library here. It doesn't have integration tests/snippets yet, but when it does, they will have the same requirements.

Reply:
Good plan, will do.